python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -69,6 +69,18 @@ export default function EditorContainer({
     [dispatch]
   );
 
+  const onUpdatePromptParameters = useCallback(
```

Comment:
I think we should have this at the single parameter granularity -- we should have both set_parameter and delete_parameter endpoints being implemented

Reply:
Can do that -- should I do that in this diff or at top of stack?

Reply:
Functionally speaking, any reason for one or the other?

Reply:
A couple things:
1. It's a more accurate mapping to the config api set/delete (single) parameter methods
2. I think it should more easily allow us to have the temporary 'empty' parameters in state in the UI here and then only send them to the config when a name/value is set

Reply:
Resolving this comment because we discussed offline and realized that per-parameter updates won't work reliably:

Suppose I have parameter1: `{key: "test", val: "value"}`
Now I'm typing parameter2, which eventually will be `{key: "test2", val: "something"}`
While typing, parameter2's key becomes `"test"` (i.e. parameter1.key is a prefix of parameter2).
I send the setParameter request at this point, which will overwrite parameter1's value.
Eventually, I finish typing, so parameter2.key is now `"test2"`. Now "oldParameterKey" = `"test"`, and the update deletes `"test"` key and just leaves `"test2"`.
So the update would have wiped out parameter1 entirely from the parameters object.

The best option is to keep updating parameters as a single update to the entire `parameters` JSON object

Reply:
Why not wait until user clicks away from the editing UI component to send the request instead of `onUpdateParameters`? It seems wasteful to dispatch a server request until user is 'finished' setting their params

Reply:
I don't think either UX changes what's needed. If we only send the request when the user clicks away we still want to send one request of all parameters and have the server set them each instead of the client waiting and setting n param update requests, right?